src/OSPSuite.Presentation/Presenters/Importer/IModalImporterPresenter.cs

```diff
@@ -21,28 +22,29 @@ public ModalImporterPresenter(IModalImporterView view) : base(view)
       }
 
       public (IReadOnlyList<DataRepository> DataRepositories, ImporterConfiguration Configuration) ImportDataSets(
-         IImporterPresenter presenter, 
+         IImporterPresenter presenter,
          string configurationId = null
       )
       {
-         IReadOnlyList<DataRepository> result = null;
+         IReadOnlyList<DataRepository> results = Array.Empty<DataRepository>();
```

Comment:
This was the bug here. The results were null upon creation but we should always return something.
The way this presenter is setup makes it very hard to test. Why is the ImporterPresenter passed a parameter and not in the constructor? This looks to me like a potential leak also

Reply:
If you see the use of it on DataImporter service, you will see that the ImporterPresenter is configured there and then passed to the ModalImporterPresenter. So, the DataImporter is responsible for handling the ImporterPresenter while the ModalImporterPresenter is just presenting it as a modal view. I am wondering if we should have a single class for both but then, DataImporter is not aware of any UI specifics, so it is just a service while ModalImporterPresenter is UI specific. This was the way to separate the service from the UI which was mixed before. So, all in all, the DataImporter is a service you could potentially (not currently) use without DevExpress and the way to present it is by means of the ModalImporterPresenter. Maybe in the future we will need the importer logic from R or CLI and we will need a different implementation of the service (DataImporter) that will do more or less the same by means of the console or by interactions with R wrappers. We could think this better later if you wish.

Reply:
I am talking bout the ImporterPresneter not the DataImporter. There are two presenters created and one is passed to the other. This is what is very weird and makes testing very hard to write also. At any rate, it works